src/components/chart/statistics-chart.ts

```diff
@@ -0,0 +1,315 @@
+          )
+        )
+      );
+
+    if (endTime > new Date()) {
+      endTime = new Date();
+    }
+
+    const names = this.names || {};
+    statisticsData.forEach((stats) => {
+      const firstStat = stats[0];
+      let name = names[firstStat.statistic_id];
+      if (!name) {
+        const entityState = this.hass.states[firstStat.statistic_id];
+        if (entityState) {
+          name = computeStateName(entityState);
+        } else {
+          name = firstStat.statistic_id;
+        }
+      }
+      // array containing [value1, value2, etc]
+      let prevValues: any[] | null = null;
+
+      const data: ChartDataset<"line">[] = [];
+
+      const pushData = (timestamp: Date, datavalues: any[] | null) => {
+        if (!datavalues) return;
+        if (timestamp > endTime) {
+          // Drop datapoints that are after the requested endTime. This could happen if
+          // endTime is "now" and client time is not in sync with server time.
+          return;
+        }
+        data.forEach((d, i) => {
+          if (datavalues[i] === null && prevValues && prevValues[i] !== null) {
+            // null data values show up as gaps in the chart.
+            // If the current value for the dataset is null and the previous
+            // value of the data set is not null, then add an 'end' point
+            // to the chart for the previous value. Otherwise the gap will
+            // be too big. It will go from the start of the previous data
+            // value until the start of the next data value.
+            d.data.push({ x: timestamp.getTime(), y: prevValues[i] });
+          }
+          d.data.push({ x: timestamp.getTime(), y: datavalues[i] });
+        });
+        prevValues = datavalues;
+      };
+
+      const addDataSet = (
+        nameY: string,
+        step = false,
+        fill = false,
+        color?: string
+      ) => {
+        if (!color) {
+          color = getColorByIndex(colorIndex);
+          colorIndex++;
+        }
+        data.push({
+          label: nameY,
+          fill: fill ? "origin" : false,
+          borderColor: color,
+          backgroundColor: color + "7F",
+          stepped: step ? "before" : false,
+          pointRadius: 0,
+          data: [],
+        });
+      };
+
+      const statTypes: this["statTypes"] = [];
+
+      this.statTypes.forEach((type) => {
+        if (statisticsHaveType(stats, type)) {
+          statTypes.push(type);
+          addDataSet(
+            `${name} (${this.hass.localize(
+              `ui.components.statistics_charts.statistic_types.${type}`
+            )})`,
+            false
+          );
+        }
+      });
+
+      let lastDate: Date;
+
+      // Process chart data.
+      stats.forEach((stat) => {
+        const value: Array<number | null> = [];
+        statTypes.forEach((type) => {
+          value.push(
+            stat[type] ? Math.round(stat[type]! * 100) / 100 : stat[type]
+          );
+        });
```

Comment:
```suggestion
        statTypes.forEach((type) => {
          if (stat[type]) {
            value.push(Math.round(stat[type] * 100) / 100);
          }
        });
```
There is no need to push these right?

Reply:
We should push `null` when the data is not available.